dali/pipeline/operators/reader/parser/coco_parser.h

```diff
@@ -52,9 +52,10 @@ using AnnotationMap = std::multimap<int, Annotation>;
 
 class COCOParser: public Parser<ImageLabelWrapper> {
  public:
-  explicit COCOParser(const OpSpec& spec, const AnnotationMap& annotations_multimap)
+  explicit COCOParser(const OpSpec& spec, const AnnotationMap& annotations_multimap, const bool save_img_ids)
     : Parser<ImageLabelWrapper>(spec),
-    annotations_multimap_(annotations_multimap) {}
+    annotations_multimap_(annotations_multimap),
+    save_img_ids_(save_img_ids) {}
```

Comment:
It is wrong as you bound temporary to reference - §12.2/5 c++ standard says that save_img_ids is invalid as soon as ctor ends.

Reply:
It should work but I wonder why `save_img_ids_` need to be a reference after all?
If someone call COCOParser constructor with some temporal variable from the stack it will also cause problems, like:
```
parser_.reset(new COCOParser(spec, annotations_multimap_, false));
```